### PR TITLE
Fix tests on ubuntu github ci

### PR DESCRIPTION
**Pull request by hiergaut**
Fix all tests which not passed on github ci

### CODE REVIEW

**Comment by hiergaut** [comment]
Waiting for #15 merged

**Comment by hiergaut** [comment]
Waiting for #21 #22 merged ...

**Comment by hiergaut** [comment]
Nice job, full test passed.

**Comment by hiergaut** [comment]
What is "build Expected — Waiting for status to be reported" check ?

**Comment by hiergaut** [comment]
@nmellado ready to merge

File: src/server/impl2/ViewerClient2.cpp [hunk 1]
```diff
@@ -100,6 +100,7 @@ void ViewerClient2::notifyNewStreamer( const StreamerClient2* streamer ) {
 
     m_clientStreamAdded = false;
     int iTry            = 0;
+    // Todo fix this
```
Comment (nmellado) [inline src/server/impl2/ViewerClient2.cpp:103]: Why is this todo for ?
It is not clear why it is wrong. Please add a description of the problem so contributors can understand what needs to be fixed.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -31,23 +30,39 @@ list(APPEND test_headers ${CMAKE_CURRENT_SOURCE_DIR}/test_common.hpp)
...
+set(LINK_LIBRARIES ${LINK_LIBRARIES} test-common)
+
+if(HUB_BUILD_CORE)
+    find_package(Eigen3 QUIET)
+    if(Eigen3_FOUND)
+        set(LINK_LIBRARIES ${LINK_LIBRARIES} Eigen3::Eigen)
+        set(INCLUDE_LIBRARIES ${INCLUDE_LIBRARIES} ${EIGEN3_INCLUDE_DIR})
+        # # message(STATUS "${HEADER_MSG} Eigen found: ${EIGEN3_INCLUDE_DIR}")
+    else()
+       list(REMOVE_ITEM test_srcs ${CMAKE_CURRENT_SOURCE_DIR}/core/matrix/test-core-matrix-compatEigen3.cpp)
```
Comment (nmellado) [inline tests/CMakeLists.txt:42]: Add a warning to prompt that this test has been disabled

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -31,23 +30,39 @@ list(APPEND test_headers ${CMAKE_CURRENT_SOURCE_DIR}/test_common.hpp)
...
-    if(NOT OpenCV_FOUND)
-        list(REMOVE_ITEM test_srcs ${CMAKE_CURRENT_SOURCE_DIR}/core/matrix/test-core-matrix-compatOpenCV.cpp)
+
+    find_package(OpenCV QUIET COMPONENTS core imgproc)
+    if(OpenCV_FOUND)
+        set(LINK_LIBRARIES ${LINK_LIBRARIES} opencv_core opencv_imgproc)
+        set(INCLUDE_LIBRARIES ${INCLUDE_LIBRARIES} ${OpenCV_INCLUDE_DIRS}/opencv2)
+        # # message(STATUS "${HEADER_MSG} OpenCV found: ${OpenCV_INCLUDE_DIRS}")
+    else()
+       list(REMOVE_ITEM test_srcs ${CMAKE_CURRENT_SOURCE_DIR}/core/matrix/test-core-matrix-compatOpenCV.cpp)
```
Comment (nmellado) [inline tests/CMakeLists.txt:51]: Add a warning to prompt that this test has been disabled

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -31,23 +30,39 @@ list(APPEND test_headers ${CMAKE_CURRENT_SOURCE_DIR}/test_common.hpp)
...
-    if(NOT OIIO_FOUND)
-        list(REMOVE_ITEM test_srcs ${CMAKE_CURRENT_SOURCE_DIR}/core/matrix/test-core-matrix-compatOIIO.cpp)
+
+    find_package(OpenImageIO QUIET)
+    if(OpenImageIO_FOUND)
+        set(LINK_LIBRARIES ${LINK_LIBRARIES} OpenImageIO)
+        set(INCLUDE_LIBRARIES ${INCLUDE_LIBRARIES} ${OpenImageIO_INCLUDE_DIR}/OpenImageIO)
+        # # message(STATUS "${HEADER_MSG} OpenImageIO found: ${OpenImageIO_INCLUDE_DIR}/OpenImageIO")
+    else()
+       list(REMOVE_ITEM test_srcs ${CMAKE_CURRENT_SOURCE_DIR}/core/matrix/test-core-matrix-compatOIIO.cpp)
```
Comment (nmellado) [inline tests/CMakeLists.txt:60]: Add a warning to prompt that this test has been disabled

File: tests/client/test-client-ViewerSensor.cpp [hunk 5]
```diff
@@ -26,7 +26,7 @@ TEST_CASE( "Viewer" ) {
         int nNewStreamer                     = 0;
         int nDelStreamer                     = 0;
         std::atomic<int> nServerNotFound     = 0;
-        int nServerConnected                 = 0;
+        std::atomic<int> nServerConnected                 = 0;
```
Comment (nmellado) [inline tests/client/test-client-ViewerSensor.cpp:29]: Spacing to `= 0;` looks suspicious.

File: tests/native/test-native-ViewerStream.cpp [hunk 6]
```diff
@@ -16,10 +16,10 @@ TEST_CASE( "Native Viewer stream" ) {
 
     {
 
-        static int nNewStreamer                     = 0;
-        static int nDelStreamer                     = 0;
+        static std::atomic<int> nNewStreamer                     = 0;
+        static std::atomic<int> nDelStreamer                     = 0;
         static std::atomic<int> nServerNotFound     = 0;
-        static int nServerConnected                 = 0;
+        static std::atomic<int> nServerConnected                 = 0;
```
Comment (nmellado) [inline tests/native/test-native-ViewerStream.cpp:22]: Spacings look suspicious here also.

File: tests/sensor/test-sensor-compatOsArch.cpp [hunk 7]
```diff
@@ -9,6 +9,7 @@
 #include <sensor/OutputSensor.hpp>
 
 TEST_CASE( "sensor compat os and arch test" ) {
+//    int main() {
```
Comment (nmellado) [inline tests/sensor/test-sensor-compatOsArch.cpp:12]: Why this new line ?

File: tests/test-wip.cpp
Comment (nmellado) [inline tests/test-wip.cpp:1]: Why this file ?

File: tests/test_common.cpp [hunk 8]
```diff
@@ -182,10 +183,10 @@ void _checkValue( double value,
                 const auto mean = algo::Mean( values.begin(), values.begin() + nEl );
                 report += "\n\t\t\033[31mvalue:" + std::to_string( value ) +
                           ", mean:" + std::to_string( mean ) +
-                          ", σ:" + std::to_string( standardDeviation ) +
-                          ", mean-σ(15.9%):" + std::to_string( mean - standardDeviation ) +
-                          ", mean-2σ(2.3%):" + std::to_string( mean - 2 * standardDeviation ) +
-                          ", mean-3σ(0.1%):" + std::to_string( mean - 3 * standardDeviation ) +
+                          ", ?:" + std::to_string( standardDeviation ) +
+                          ", mean-?(15.9%):" + std::to_string( mean - standardDeviation ) +
```
Comment (nmellado) [inline tests/test_common.cpp:187]: Why using `?` here instead of sigma ?

File: tests/test_common.hpp [hunk 9]
```diff
@@ -204,7 +207,7 @@ void _checkValue( double value,
                   const std::string& name,
                   const std::string& unit,
                   const std::string& filename,
-                  int line );
+                         int line );
```
Comment (nmellado) [inline tests/test_common.hpp:210]: Spacing change looks suspicious

**Comment by hiergaut** [comment]
All conversations resolved for me, you can merged @nmellado

**Review by nmellado — APPROVED** [review]

**Pull request merged**